### ISSUE TITLE
Error: "failed to move sink input" for specific application

### ISSUE THREAD

**Issue by railhoff2** [wwmm/pulseeffects#422]
I've got a fairly easily reproducible error, which happens with the application I am using (qTox) but not with Firefox. I'm brand new to pulseeffects so these are the only two programs I have tested thus far. I've tested it in a fresh install even, to try to isolate the problem.

The problem is that when the qTox application creates its audio output stream that pulse can see, this error is shown in pulseeffects:

> (pulseeffects:2556): pulseeffects-CRITICAL **: 11:51:31.678: pulse_manager: failed to move sink input: qtox, idx = 2 to PE
> 
> (pulseeffects:2556): pulseeffects-CRITICAL **: 11:51:31.684: pulse_manager: failed to move source output qtox, idx =  to PE

The box for qTox's output can be "checked", but the graph showing the application audio output at the top of the program has no activity. Each time you try to check the box, another error is printed out. Everything works as expected when working with Firefox audio output.

Exact steps to reproduce (qTox specific parts should take less than 1 minute to complete, no email or sign up needed):

1. Install Linut Mint 19.1 MATE 64 bit on a machine, or in a VM (which I did).
2. Download the qTox Appimage: https://github.com/qTox/qTox/releases/download/v1.16.3/qTox-v1.16.3.x86_64.AppImage
3. Install pulseeffects via the PPA, including installing the recommended packages, or use the flatpack, both will do the same thing.
4. Reboot, if desired, this step isn't needed.
5. Start pulseeffects.
6. chmod +x ~/Downloads/qTox-v1.16.3.x86_64.AppImage && ~/Downloads/qTox-v1.16.3.x86_64.AppImage
-create a local account, which is just a username and an optional password, takes less than 5 seconds.
-one qTox is loaded, hit the gear at the bottom of the grey area and navigate to "Audio/Video"
7. At this point, pulseeffects should display the errors on the command line:
> (pulseeffects:2556): pulseeffects-CRITICAL **: 11:51:31.678: pulse_manager: failed to move sink input: qtox, idx = 2 to PE
> 
> (pulseeffects:2556): pulseeffects-CRITICAL **: 11:51:31.684: pulse_manager: failed to move source output qtox, idx =  to PE

8. If you don't see the errors printed out, uncheck and then check the box labelled "Test Audio", in the "Audio/Video" tab. This is probably needed if you don't have a microphone that the VM can access, since this makes the qTox application output a sound.

Note that the focus of this bug, for me, at least, is the desire to modify the incoming audio from the other person I am speaking with, not to add effects to my own microphone. Just to reduce the static coming in from the person I am speaking to.

Most oddly, I was able to get qTox to work with pulse-effects exactly one time. It was about my 10th time launching the program, but in the next ~30 tries I have never been able to get it working with pulseeffects again. I have no idea why, but it worked GREAT that one single time, wanting to get the benefits of it again!

Pulseeffects versions tested: 4.4.6 from the PPA, and the 4.3.8 flatpak.

**Comment by railhoff2** [comment]
One more thing to add: the only other output that pulseeffects prints out is this, which happens right when it starts up:
> (pulseeffects:2635): pulseeffects-WARNING **: 11:52:07.351: delay plugin was not found!
This is just a warning, so I'm assuming it isn't related.

**Comment by wwmm** [comment]
The `delay plugin was not found` is happening because you do not have `Linux Studio Plugins` installed. But this is only a problem in case you want to use the delay plugin. This is optional. No need to worry.

The error `failed to move sink input` also happens with `Telegram` as you can see in #222. As that discussion is very long I suggest you use `Ctrl-F` to find `telegram`. In any case at least for Telegram the bug is on its side as you can see here https://github.com/telegramdesktop/tdesktop/issues/2231. I don't know how they did this but somehow they are forcing Pulseaudio to not accept Telegram audio to be routed to a sink that is not the default. My guess is that the same is happening with qTox

**Comment by railhoff2** [comment]
Thanks for the quick response. You're right. When I go in to pavucontrol, I notice that I cannot change the output device for qTox, but I can watch it change for Firefox as I use pulseeffects with Firefox.

Luckily, I can change the audio output device inside of qTox as a workaround. Once I changed that to "pulseEffects(apps)", pulseeffects started working with qTox! I was able to equalize the audio for my output stream, as well as for my microphone.

It makes a WORLD of difference. Painful static is now hardly noticeable at all.

I'll share the setting I used to make this improvement, in case anyone is curious. I started with the "Classical" preset, and then moved the sliders as best I could to capture the human male vocal range of 85-185 hz, or so I hear:
![pulseeffects](https://user-images.githubusercontent.com/47122065/51876921-108ac900-231f-11e9-97b7-38fbe2502ec3.png)
